pkg/envoy/rds/response.go

```diff
@@ -49,30 +49,32 @@ func NewResponse(catalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_disco
 		isSourceService := trafficPolicy.Source.Equals(proxyServiceName)
 		isDestinationService := trafficPolicy.Destination.Equals(proxyServiceName)
 		svc := trafficPolicy.Destination
-		hostnames, err := catalog.GetResolvableHostnamesForUpstreamService(proxyServiceName, svc)
-		//filter out traffic split service, reference to pkg/catalog/xds_certificates.go:74
-		if isTrafficSplitService(svc, allTrafficSplits) {
-			continue
-		}
-		if err != nil {
-			log.Error().Err(err).Msg("Failed listing domains")
-			return nil, err
-		}
 		weightedCluster, err := catalog.GetWeightedClusterForService(svc)
 		if err != nil {
 			log.Error().Err(err).Msg("Failed listing clusters")
 			return nil, err
 		}
 
-		for _, hostname := range hostnames {
-			// All routes from a given source to destination are part of 1 traffic policy between the source and destination.
-			for _, httpRoute := range trafficPolicy.HTTPRouteMatches {
-				if isSourceService {
-					aggregateRoutesByHost(outboundAggregatedRoutesByHostnames, httpRoute, weightedCluster, hostname)
-				}
+		if weightedCluster.Weight > 0 {
```

Comment:
Inverting this condition to `<= 0` would avoid extra level of nesting.

Reply:
I'm also a huge fan of avoiding unnecessary nesting. @snehachhabria would you be so kind to make another PR after this?

Reply:
@draychev and @shashankram I have updated the code, PTAL